pkg/tbtc/tbtc.go

```diff
@@ -56,19 +56,22 @@ func Initialize(
 	workPersistence persistence.BasicHandle,
 	scheduler *generator.Scheduler,
 	config Config,
-	registry *diagnostics.Registry,
+	metricsRegistry *metrics.Registry,
 ) error {
 	node := newNode(chain, netProvider, keyStorePersistence, workPersistence, scheduler, config)
 	deduplicator := newDeduplicator()
 
-	registry.RegisterApplicationSource(
-		"tbtc",
-		func() map[string]interface{} {
-			return map[string]interface{}{
-				"preParamsPoolSize": node.dkgExecutor.PreParamsCount(),
-			}
-		},
-	)
+	if metricsRegistry != nil {
+		// only if metrics are configured
+		metricsRegistry.ObserveApplicationSource(
+			"tbtc",
+			map[string]metrics.Source{
+				"pre_params_pool_size": func() float64 {
```

Comment:
The naming could be confusing. We've got `pre_params_pool_size` here for a metric with the current size of the pool. And in the config, we've got `PreParamsPoolSize` property to specify the target size for the pool.

It would be good to rename the config property (and flag) to `preParamsPoolTarget` or `preParamsPoolTargetSize`.

Reply:
Agreed. We have two options:

Option 1:
Pre-params pool size + Number of pre-params in pool

Option 2:
Pre-params target pool size + Pre-params pool size.

I slightly lean toward Option 1. WDYT?

Reply:
1 is fewer changes. So let's go with 1

Reply:
c71ec67b7fc2443c0e3f2e92260fab25873da97d